### PR TITLE
[Merged by Bors] - fetcher: reduce multierror log

### PR DESCRIPTION
**Pull request by countvonzero**
## Motivation
@pigmej experienced some data corruption while syncing to a devnet that has a different data structure for ATX.
his log blew up with messages like below. it eventually OOM'ed. 

this is because the logic of data availability is recursive. even though fetcher already offload the data validation to a separate goroutine, effectively make each call stack 2 layers deep, the logs still scale to the number of hashes fetched, and the data it references. this PR aims to reduce memory pressure in such case (incompatible data).

```
2023-02-28T10:00:45.673+0100	WARN	fbdc7.app.fbdc7.proposalListener	ballot data availability check failed	{"node_id": "fbdc703ddaa6ccc1d96303e01c3135c46925c8331f922ed1c75152a628b1b379", "module": "app", "node_id": "fbdc703ddaa6ccc1d96303e01c3135c46925c8331f922ed1c75152a628b1b379", "module": "proposalListener", "sessionId": "51b514a6-fc0a-4029-835e-b689341af85b", "ballot_id": "d2ee71f09d", "layer_id": 127, "errmsg": "fetch ballots: 2 errors occurred:\n\t* decode ballot f9296e13ec: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\n\n\n\n\n\n\n\n\n\n\n\n\n", "name": "proposalListener"}
2023-02-28T10:00:45.673+0100	WARN	fbdc7.app.fbdc7.fetcher	failed to get hash	{"node_id": "fbdc703ddaa6ccc1d96303e01c3135c46925c8331f922ed1c75152a628b1b379", "module": "app", "node_id": "fbdc703ddaa6ccc1d96303e01c3135c46925c8331f922ed1c75152a628b1b379", "module": "fetcher", "sessionId": "51b514a6-fc0a-4029-835e-b689341af85b", "hint": "ballotDB", "hash": "0x3a44c726f575f9a42372085e5d68d5a54ed1c2c0000000000000000000000000", "errmsg": "fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* fetch ballots: 2 errors occurred:\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\t* decode ballot 75652dc112: read atx for ballot height: get ATXs from DB: decode value 0 is out of range\n\n\n\n\n\n\n\n\n\n\n\n", "name": "fetcher"}
```

## Changes
- reduce logging to debug for the actual multierror
- remove error from the warning log

### CODE REVIEW

File: fetch/mesh_data.go [hunk 1]
```diff
@@ -48,18 +48,26 @@ func (f *Fetch) getHashes(ctx context.Context, hashes []types.Hash32, hint datas
...
 	}
 
-	return eg.Wait().ErrorOrNil()
+	err := eg.Wait().ErrorOrNil()
+	if err != nil {
+		// due to the recursive nature of data fetching in go-spacemesh, we want to minimize the log volume
+		f.logger.WithContext(ctx).With().Debug("failed to get hashes",
+			log.String("hint", string(hint)),
+			log.Err(err))
+	}
```
Comment (poszu) [inline fetch/mesh_data.go]: Why log it at all here? IMO the code that handles the error should log it. Any contextual information like `hint` could be backed into the returned error.

Comment (countvonzero) [reply fetch/mesh_data.go]: removed. thanks.

**Review by poszu — APPROVED** [review]

**Comment by countvonzero** [comment]
bors merge

**Comment by countvonzero** [comment]
bors merge

**Pull request closed without merging**